lib/buffer.js

```diff
@@ -42,34 +42,92 @@ function alignPool() {
   }
 }
 
-
-function Buffer(arg, encoding) {
+/**
+ * The Buffer() construtor is "soft deprecated" ... that is, it is deprecated
+ * in the documentation and should not be used moving forward. Rather,
+ * developers should use one of the three new factory APIs: Buffer.from(),
+ * Buffer.allocUnsafe() or Buffer.alloc() based on their specific needs. There
+ * is no hard deprecation because of the extent to which the Buffer constructor
+ * is used in the ecosystem currently -- a hard deprecation would introduce too
+ * much breakage at this time. Eventually, however, we should re-evaluate that
+ * choice.
```

Comment:
@trevnorris I thought we could override the default ES6 subclass behavior using `@@species`? For example, @littledan sent this PR to feross/buffer that makes the Uint8Array methods call the `Uint8Array` constructor instead of the `Buffer` one: https://github.com/feross/buffer/pull/97 Was there something new that came up that is preventing this strategy?